### PR TITLE
Created new tests for OpCollection*

### PR DESCRIPTION
**Pull request by wsuchy**
Added new tests for `OpCollection*` that extends `OpTransformerSpec`. 
I've kept original tests as they are more detailed.

### CODE REVIEW

File: core/src/test/scala/com/salesforce/op/stages/impl/feature/OPMapTransformerTest.scala [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+/**
+ * @author ksuchanek
+ * @since 214
+ */
+@RunWith(classOf[JUnitRunner])
+class OPMapTransformerTest extends OpTransformerSpec[IntegralMap, TransformerType] {
+
+  import OPMapTransformerTest._
+
+  lazy val (dataEmailMap, top) = TestFeatureBuilder("name",
```
Comment (tovbinm) [inline core/src/test/scala/com/salesforce/op/stages/impl/feature/OPMapTransformerTest.scala]: `lazy val (inputData, top) = ...`

File: core/src/test/scala/com/salesforce/op/stages/impl/feature/OPMapTransformerTest.scala [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    IntegralMap(Map("p1" -> 10L, "p2" -> 11L))
+  )
+}
+
+object OPMapTransformerTest {
+  type TransformerType = OPMapTransformer[Email, Integral, EmailMap, IntegralMap]
+
+  class BaseTransformer extends UnaryTransformer[Email, Integral](
+    operationName = "testUnary",
+    uid = "1234"
```
Comment (tovbinm) [inline core/src/test/scala/com/salesforce/op/stages/impl/feature/OPMapTransformerTest.scala]: `uid = UID[BaseTransformer]`

**Comment by wsuchy** [comment]
@tovbinm  I pushed changes regarding your comments. The companion objects are necessary because of the underlying lambdas and this is the easiest way of making serializing/deserializing working.

**Comment by tovbinm** [comment]
@wsuchy I dont see any lambdas. these are just regular transformers. simply define them in the same file as the test. I.e. this should work:
```scala
@RunWith(classOf[JUnitRunner])
class OPListTransformerTest extends OpTransformerSpec[TextList, BaseTransformer] {
  // ... 
  val transformer new = BaseTransformer()
  // ... 
}

class BaseTransformer extends UnaryTransformer[Text, Text](
  operationName = "testUnary",
  uid = UID[BaseTransformer]
) {
  override def transformFn: (Text => Text) = (input: Text) => input.value.map(_.toLowerCase()).toText
}

```

**Pull request merged**